src/main/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecution.java

```diff
@@ -712,17 +712,26 @@ public void onLoad(FlowExecutionOwner owner) throws IOException {
         this.owner = owner;
 
         try {
+            initializeStorage();  // Throws exception and bombs out if we can't load FlowNodes
+        } catch (Exception ex) {
             try {
-                initializeStorage();  // Throws exception and bombs out if we can't load FlowNodes
-            } catch (Exception ex) {
-                LOGGER.log(Level.WARNING, "Error initializing storage and loading nodes, will try to create placeholders for: "+this, ex);
+                if (!canResume()) {
```

Comment:
Alternatively, we could check this even before we attempt to initialize storage and just bail out right away. I don't know if that is too pessimistic, maybe there is a case where the Pipeline is not resumable but flow nodes are all up to date and so the optimistic behavior here is useful because you will still be able to see the existing flow graph in those cases?